hardware/Meteorologisk.cpp

```diff
@@ -168,13 +167,20 @@ void CMeteorologisk::Do_Work()
 		}
 		if (sec_counter % Meteorologisk_Poll_Interval == 0)
 		{
-			try
+			if(!m_URL.empty())
 			{
-				GetMeterDetails();
+				try
+				{
+					GetMeterDetails();
+				}
+				catch (...)
+				{
+					Log(LOG_ERROR, "Unhandled failure getting/parsing data!");
+				}
 			}
-			catch (...)
+			else
 			{
-				Log(LOG_ERROR, "Unhandled failure getting/parsing http data!");
+				Log(LOG_STATUS, "Unable to perform work due to missing/incorrect startup parameters!");
```

Comment:
This message is a bit unclear... imaging you are looking at this 1 day later... you totally missed the startup ...
I think it's clearer to log here 

"Unable to properly start due to missing or incorrect Location parameters (Latitude, Longitude)!"

(And remove it from StartHardware)

As **sec_counter** is initialized at 290 (I would initialize it to to **Meteorologisk_Poll_Interval -5**)
This log message is quickly displayed

Reply:
Also done :)